Sources/XiEditor/LineCache.swift

```diff
@@ -17,44 +17,39 @@ import Foundation
 /// A half-open range representing lines in a document.
 typealias LineRange = CountableRange<Int>
 
-/// Represents a single line, including rendering information.
 struct Line<T> {
     var text: String
     var cursor: [Int]
     var styles: [StyleSpan]
     /// This line's logical number, if it is the start of a logical line
     var number: UInt?
-    /// Associated data, to be managed by client
-    var assoc: T?
 
     /// A Boolean indicating whether this line contains a cursor.
     var containsCursor: Bool {
         return cursor.count > 0
     }
 
-    init(fromJson json: [String: AnyObject]) {
-        // this could be a more clear exception
-        text = json["text"] as! String
-        cursor = json["cursor"] as? [Int] ?? []
-        number = json["ln"] as? UInt
-        if let styles = json["styles"] as? [Int] {
-            self.styles = StyleSpan.styles(fromRaw: styles, text: self.text)
-        } else {
-            self.styles = []
-        }
-    }
+    /// Associated data, to be managed by client
+    var assoc: T?
+}
 
-    /// Create a new line, applying new styles to an existing line's text
-    init?(updateFromJson line: Line?, json: [String: AnyObject]) {
-        guard let line = line else { return nil }
+extension Line {
```

Comment:
Moving this to an extension causes Swift to auto-generate a memberwise initializer.

https://docs.swift.org/swift-book/LanguageGuide/Initialization.html#ID214